Could not deploy docker app resources

When using app resource URI as docker URL, the stream deployment fails as it could not load the docker resource file.

To reproduce:

> dataflow:>app register --type source --name time --uri docker:springcloudstream/time-source-kafka:latest
> dataflow:>app register --type sink --name log --uri docker:springcloudstream/log-sink-kafka:latest
> dataflow:>stream create a1 --definition "time | log" --deploy

Following exception is thrown:

```
java.lang.RuntimeException: Exception trying to list configuration properties for application ServletContext resource [/docker:springcloudstream/log-sink-kafka:latest]
    at org.springframework.cloud.dataflow.configuration.metadata.ApplicationConfigurationMetadataResolver.listProperties(ApplicationConfigurationMetadataResolver.java:123) ~[spring-cloud-dataflow-configuration-metadata-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.cloud.dataflow.server.controller.StreamDeploymentController.qualifyParameters(StreamDeploymentController.java:267) ~[spring-cloud-dataflow-server-core-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.cloud.dataflow.server.controller.StreamDeploymentController.deployStream(StreamDeploymentController.java:243) ~[spring-cloud-dataflow-server-core-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.cloud.dataflow.server.controller.StreamDeploymentController.deploy(StreamDeploymentController.java:186) ~[spring-cloud-dataflow-server-core-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_20]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_20]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_20]
    at java.lang.reflect.Method.invoke(Method.java:483) ~[na:1.8.0_20]
    at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221) ~[spring-web-4.2.5.RELEASE.jar!/:4.2.5.RELEASE]
    at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136) ~[spring-web-4.2.5.RELEASE.jar!/:4.2.5.RE
...
...
Caused by: java.lang.IllegalArgumentException: File must exist
    at org.springframework.boot.loader.data.RandomAccessDataFile.<init>(RandomAccessDataFile.java:67) ~[spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.data.RandomAccessDataFile.<init>(RandomAccessDataFile.java:51) ~[spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.jar.JarFile.<init>(JarFile.java:95) ~[spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.archive.JarFileArchive.<init>(JarFileArchive.java:65) ~[spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.archive.JarFileArchive.<init>(JarFileArchive.java:61) ~[spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
    at org.springframework.cloud.dataflow.configuration.metadata.ApplicationConfigurationMetadataResolver.listProperties(ApplicationConfigurationMetadataResolver.java:87) ~[spring-cloud-dataflow-configuration-metadata-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
```

--- a/pom.xml
+++ b/pom.xml
@@ -26,7 +26,7 @@
 		<java.version>1.8</java.version>
 		<spring-cloud-dataflow-ui.version>1.1.0.RELEASE</spring-cloud-dataflow-ui.version>
 		<spring-cloud-deployer.version>1.1.1.RELEASE</spring-cloud-deployer.version>
-		<spring-cloud-deployer-local.version>1.1.1.RELEASE</spring-cloud-deployer-local.version>
+		<spring-cloud-deployer-local.version>1.1.2.BUILD-SNAPSHOT</spring-cloud-deployer-local.version>
 
 		<!-- Note: Change ./spring-cloud-dataflow-server-local/pom.xml to match the spring-cloud-task.version -->
 		<spring-cloud-task.version>1.1.0.RELEASE</spring-cloud-task.version>

--- a/spring-cloud-dataflow-server-local/pom.xml
+++ b/spring-cloud-dataflow-server-local/pom.xml
@@ -19,6 +19,7 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 		<java.version>1.8</java.version>
+		<spring-cloud-deployer.version>1.1.1.RELEASE</spring-cloud-deployer.version>
 		<spring-cloud-task.version>1.1.0.RELEASE</spring-cloud-task.version>
 		<spring-cloud.version>Camden.SR2</spring-cloud.version>
 	</properties>
@@ -68,11 +69,15 @@
 		</developer>
 	</developers>
 
-
 	<dependencies>
 		<dependency>
 			<groupId>org.springframework.cloud</groupId>
 			<artifactId>spring-cloud-starter-dataflow-server-local</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-deployer-resource-docker</artifactId>
+			<version>${spring-cloud-deployer.version}</version>
 		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>

--- a/spring-cloud-dataflow-server-local/src/main/java/org/springframework/cloud/dataflow/server/local/LocalDataFlowServer.java
+++ b/spring-cloud-dataflow-server-local/src/main/java/org/springframework/cloud/dataflow/server/local/LocalDataFlowServer.java
@@ -18,13 +18,16 @@ package org.springframework.cloud.dataflow.server.local;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.cloud.dataflow.server.EnableDataFlowServer;
+import org.springframework.context.annotation.Import;
 
 /**
  * Bootstrap class for the local Spring Cloud Data Flow Server.
  *
  * @author Mark Fisher
+ * @author Ilayaperumal Gopinathan
  */
 @EnableDataFlowServer
+@Import(LocalDataFlowServerConfiguration.class)
 public class LocalDataFlowServer {
 
 	public static void main(String[] args) {

--- a/spring-cloud-dataflow-server-local/src/main/java/org/springframework/cloud/dataflow/server/local/LocalDataFlowServerConfiguration.java
+++ b/spring-cloud-dataflow-server-local/src/main/java/org/springframework/cloud/dataflow/server/local/LocalDataFlowServerConfiguration.java
@@ -1,0 +1,45 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.local;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.cloud.deployer.resource.docker.DockerResourceLoader;
+import org.springframework.cloud.deployer.resource.maven.MavenProperties;
+import org.springframework.cloud.deployer.resource.maven.MavenResourceLoader;
+import org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ResourceLoader;
+
+/**
+ * Configuration class for Local Data Flow Server.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class LocalDataFlowServerConfiguration {
+
+	@Bean
+	public DelegatingResourceLoader delegatingResourceLoader(MavenProperties mavenProperties) {
+		DockerResourceLoader dockerLoader = new DockerResourceLoader();
+		MavenResourceLoader mavenResourceLoader = new MavenResourceLoader(mavenProperties);
+		Map<String, ResourceLoader> loaders = new HashMap<>();
+		loaders.put("docker", dockerLoader);
+		loaders.put("maven", mavenResourceLoader);
+		return new DelegatingResourceLoader(loaders);
+	}
+}

